src/Model/Behavior/SearchBehavior.php

```diff
@@ -167,19 +159,23 @@ protected function _extractParams($params, $filters)
      * ```
      *
      * @param array $params The parameters array to flatten.
-     * @param array $flattenWhitelist Keys to avoid flattening.
+     * @param array $filters The array of filters with configuration
      * @return array The flattened parameters array.
      */
-    protected function _flattenParams($params, $flattenWhitelist)
-    {
+    protected function _flattenParams($params, $filters)
+    {  
         $flattened = [];
         foreach ($params as $key => $value) {
-            if (is_array($value) && !in_array($key, $flattenWhitelist)) {
-                foreach ($value as $childKey => $childValue) {
-                    if (!is_numeric($childKey)) {
-                        $flattened[$key . '.' . $childKey] = $childValue;
-                    } else {
-                        $flattened[$key][$childKey] = $childValue;
+            if (is_array($value)) {
+                if (!empty($filters[$key]) && $filters[$key]->getConfig()['flatten'] == false) {
+                    $flattened[$key] = $value;
+                } else {
```

Comment:
because you dont bail early where you can :)

e.g. if (is_array($value)) {} could be

    if (!is_array($value)) {
        continue;
    }

    foreach ....